src/main/scala/strawman/collection/Seq.scala

```diff
@@ -26,6 +31,9 @@ trait SeqOps[+A, +CC[X], +C] extends Any
   with ArrayLike[A]
   with Equals {
 
+  protected[this] def c: C
+  protected[this] def seq: Seq[A]
```

Comment:
How do you write generic code that does seq-like things but works on `Array` and `String` too?

If users can do it that way (whatever that way is), we should just do it that way too and not try for this inheritance scheme.  Just inherit the parts that always work and reimplement the rest (possibly by deferring to a wrapping collection instead of an ops collection?).